lib/ansible/modules/system/rc_service.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/python
+
+# (c) Mattias Lindvall
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community',
+}
+
+DOCUMENTATION = '''
+module: rc_service
+author:
+    - "Mattias Lindvall"
+version_added: "2.4"
+short_description: Manages services.
+description:
+    - Manages services with rc style init scripts, and sysrc based configuration.
+options:
+    name:
+        required: true
+        description:
+            - Name of the service.
+        aliases: ['unit', 'service']
+    state:
+        required: false
+        default: null
+        choices: ['started', 'stopped', 'restarted']
+        description:
+            - C(started)/C(stopped) are idempotent actions that will not run commands unless necessary.
+              C(restarted) will always restart the service.
+    enabled:
+        required: false
+        default: null
+        choices: ['yes', 'no']
+        description:
+            - Whether the service should be enabled using sysrc.
```

Comment:
I would like to suggest a different approach: We remove support for `enabled` from this service module, and do not attempt to set any sysrc variables at all. Instead we add another module called `sysrc`, letting the user set any configuration they like. This will greatly simplify the modules and separate the concerns, and it would open up for making this service module portable across BSD's, as sysrc is only used on FreeBSD. This is what I am doing locally now, as this makes the most sense for me. What do you think?